src/OrderedListenerProvider.php

```diff
@@ -39,25 +39,47 @@ public function getListenersForEvent(object $event): iterable
         }
     }
 
+    /**
+     * Tries to get the type of a callable listener.
+     * If unable, throws an exception with information about the listener whose type could not be fetched.
+     * @param callable $listener
+     * @return string
+     */
+    private function getType(callable $listener)
```

Comment:
```suggestion
    protected function getType(callable $listener)
```

I almost never use private, only protected.  It's a useful escape hatch at times.

Reply:
That's valid. I typically use 'private' whenever a method is an implementation detail rather than part of an intended extension API, but I'll defer on that going forward.